File: assets/scripts/pingpongscale.py
import dengine.common as common
import dengine.scene as scene
import dengine.timer as timer
import math as math

def update(entity):
    current = timer.get_current()
    pingpong = math.sin(current / 1000.0)
    entity.transform.scale.z = math.fabs(pingpong) * 1.3
    entity.transform.scale.y = math.fabs(pingpong) * 0.3

File: lib/dengine-script/src/nsl-test/pingpongscale.c
#include <dengine/dengine.h>
#include <math.h>
#include <stdlib.h>

int pingpongscale_update(Entity* entity)
{
    double current = dengineutils_timer_get_current();
    double pingpong = sin(current / 1000.0);
    entity->transform.scale[2] = fabs(pingpong) * 1.3;
    entity->transform.scale[1] = fabs(pingpong) * 0.3;
    return 0;
}
